nntrainer/tensor/tensor.cpp

```diff
@@ -633,9 +633,33 @@ Tensor &Tensor::dot(Tensor const &m, Tensor &result, bool trans,
   float *rdata = result.getData();
   const float alpha = 1.0f;
   const float beta = 0.0f;
-
   enum CBLAS_TRANSPOSE transA = trans ? CblasTrans : CblasNoTrans;
   enum CBLAS_TRANSPOSE transB = trans_m ? CblasTrans : CblasNoTrans;
+
+  /// shortcut calculation in case of vector
+  /// case1: (1 * K) X (K * 1)
+  if (M == 1 && N == 1) {
+    *rdata = sdot(K, data, 1, mdata, 1);
+    return result;
+  }
+
+  /// case2: (M * K) X (K * 1)
+  if (N == 1) {
```

Comment:
how about making these all else if and lastly else with just 1 return statement?

Reply:
Done